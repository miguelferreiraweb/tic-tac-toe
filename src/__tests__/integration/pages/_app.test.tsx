import { render, screen } from '@testing-library/react';
import mockRouter from 'next-router-mock';
import { describe, expect, it } from 'vitest';

import MyApp from '@/pages/_app';

describe('App', () => {
  const MockPage = () => <div>mock page</div>;
  it('should render app unchanged', () => {
    const { container } = render(
      <MyApp Component={MockPage} pageProps={{}} router={mockRouter as any} />
    );
    expect(screen.getByText('mock page')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
