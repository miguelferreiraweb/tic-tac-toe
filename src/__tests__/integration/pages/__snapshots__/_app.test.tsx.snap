// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > should render app unchanged 1`] = `
<div>
  <div>
    mock page
  </div>
</div>
`;
